### PR TITLE
Improvements around typescript

### PR DESCRIPTION
**Pull request by samccone**
1. enforce no implicit any
2. add a note about compiling the TS before deving

### CODE REVIEW

File: lighthouse-cli/types/types.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+interface AuditResult {
+  displayValue: string;
+  debugString: string;
+  comingSoon?: boolean;
+  score: number;
+  description: string;
+  extendedInfo?: {
```
Comment (ebidel) [inline lighthouse-cli/types/types.ts]: This is missing helpText and possibly others

File: lighthouse-cli/types/types.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+interface AggregationResultItem {
+  overall: number;
+  name: string;
+  scored: boolean;
+  subItems: Array<AuditResult | string>;
+}
+
+interface Aggregation {
+  name: string;
+  score:  Array<AggregationResultItem>;
```
Comment (ebidel) [inline lighthouse-cli/types/types.ts]: Extra space

File: lighthouse-cli/types/types.ts (hunk 1 above)
Comment (samccone) [inline lighthouse-cli/types/types.ts]: yeah I am purposely keeping the type definitions here only as specific as we need within the cli. If we were to pull these typedefs up into core I will be sure to fully match the reality of the objects

File: lighthouse-cli/types/types.ts (hunk 2 above)
Comment (samccone) [inline lighthouse-cli/types/types.ts]: fixed

**Comment by samccone** [comment]
ugh not sure how I closed this... very odd.

**Pull request merged**